gravitee-apim-portal-webui-next/src/app/api-details/api-tab-subscriptions/api-tab-subscriptions.component.ts

```diff
@@ -14,35 +14,39 @@
  * limitations under the License.
  */
 import { AsyncPipe } from '@angular/common';
-import { Component, inject, Input, OnInit } from '@angular/core';
+import { Component, inject, Input } from '@angular/core';
+import { FormControl, ReactiveFormsModule } from '@angular/forms';
+import { MatFormField, MatLabel } from '@angular/material/form-field';
 import { MatIcon } from '@angular/material/icon';
+import { MatOption, MatSelect } from '@angular/material/select';
 import { MatTableModule } from '@angular/material/table';
-import { catchError, map, Observable } from 'rxjs';
+import { catchError, map, Observable, startWith, switchMap } from 'rxjs';
 import { of } from 'rxjs/internal/observable/of';
 
-import { SubscriptionMetadata, SubscriptionData } from '../../../entities/subscription/subscription';
+import { SubscriptionMetadata, SubscriptionData, Subscriptions } from '../../../entities/subscription/subscription';
 import { CapitalizeFirstPipe } from '../../../pipe/capitalize-first.pipe';
 import { SubscriptionService } from '../../../services/subscription.service';
 
 @Component({
   selector: 'app-api-tab-subscriptions',
   standalone: true,
-  imports: [AsyncPipe, MatTableModule, MatIcon, CapitalizeFirstPipe],
   providers: [CapitalizeFirstPipe],
+  imports: [AsyncPipe, MatTableModule, MatIcon, CapitalizeFirstPipe, MatFormField, MatLabel, MatSelect, ReactiveFormsModule, MatOption],
   templateUrl: './api-tab-subscriptions.component.html',
   styleUrl: './api-tab-subscriptions.component.scss',
 })
-export class ApiTabSubscriptionsComponent implements OnInit {
+export class ApiTabSubscriptionsComponent {
   @Input()
   apiId!: string;
 
-  subscriptionsList$!: Observable<SubscriptionData[]>;
-  displayedColumns: string[] = ['application', 'plan', 'status', 'expand'];
-  private subscriptionService = inject(SubscriptionService);
+  public displayedColumns: string[] = ['application', 'plan', 'status', 'expand'];
+  public subscriptionStatusesList = Subscriptions.statuses;
+  public subscriptionsStatus = new FormControl();
+  public subscriptionsList$!: Observable<SubscriptionData[]>;
 
-  constructor(private capitalizeFirstPipe: CapitalizeFirstPipe) {}
+  private subscriptionService = inject(SubscriptionService);
 
-  ngOnInit(): void {
+  constructor(private capitalizeFirstPipe: CapitalizeFirstPipe) {
```

Comment:
What was your reasoning to have the service injected above the constructor and the pipe injected within the constructor?

Reply:
Fixed, I moved one into constructor as well